submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,102 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+
+async function fetchAsync () {
+    let response = await fetch('./data/data.json');
+    let data = await response.json();
+    return data;
+}   
+  
+fetchAsync()
+    .then(data => {
+        let details = createHtml(data);
+        drawPage(details);
+        toggleTabs();
+        animation();
+    })
+    .catch(reason => console.log(reason.message));
+
+function createHtml(data) {
+    const menuItems = document.createDocumentFragment();
+    const heroItems = document.createDocumentFragment();
+    data.pancakes.forEach((elem, index) => {
+        const navContent = generateNavContent(elem, index); 
+        const heroContent = generateHeroContent(elem, index); 
+        menuItems.appendChild(navContent);
+        heroItems.appendChild(heroContent);
+    });
+    return {
+        menuItems,
+        heroItems
+    };
+}
+
+function generateNavContent(elem, index) {
```

Comment:
Why do you create elements with different approaches? I see you generated `li` elements by using `document.createElement` and links you created by using template strings.
I recommend you to use only one approach. For example, template strings.

Reply:
My idea was to show that I understand and can work with different approaches. But okay, I will bring everything to the one kind.